plugins/snowflake/credentials.go

```diff
@@ -0,0 +1,110 @@
+package snowflake
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func Credentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.Credentials,
+		DocsURL: sdk.URL("https://docs.snowflake.com/en/user-guide/snowsql.html"),
```

Comment:
I can definitely make the docs URL more specific. However, since the URL you linked doesn't mention much about the account (e.g. where to find the account name), which is an important component of the credentials, I feel like this URL might be more appropriate: https://quickstarts.snowflake.com/guide/getting_started_with_snowsql/index.html.

Just want to make sure, are you suggesting that we add one of the suggested (e.g. `UserCredentials`, `UserPassword`, `UsernamePassword`, `UserLogin`) as an addition to the `credname` package and then use that new credname in this file? If this is the case, I feel like this may be better addressed in a follow-up issue, please let me know what you think.

Reply:
Sounds good. With this in mind, I'll be resolving this thread. 

I'll make a note to open an internal ticket based on our discussion above.

Reply:
I'm not sure from the top of my head if changing the credential name later will be a breaking change. I personally think that calling this credential type `credname.UsernamePassword` would make the most sense. I believe we don't have a constant for that yet in the `credname` package, but that's an easy change. As its string value, I'd go for `"Username/Password"`.

But open to other suggestions as well.

Reply:
Since the credentials for this plugin actually contain three fields (account, username, password), I feel like just `"Username/Password"` may be a bit misleading. For this case, I feel like just `Credentials` or maybe `UserCredentials` would be best, since something like `AccountUsernamePassword` would be messy. However, I'm not sure what difference `Credentials` and `UserCredentials` would make, so I feel like keeping it as `Credentials` for now is probably fine? Let me know if you disagree.

Reply:
"Username/Password" is the means of authentication here. Does not necessarily mean that the item only contains those fields.

You can think of the credential type name as an answer to "How would you like to authenticate?" or "Which kind of credential should be used to authenticate?". Answering that with just "Credentials" seems off there, because Snowflake supports multiple different credential types (key file, token, password).

Reply:
I am proposing "Username and Password" instead of the slash naming because it makes the change straightforward to introduce without changing the logic for creating files from the credentials name during the `make new-plugin` command (slashes are not allowed in file names). @williamhpark encountered some issue with this.

Reply:
Probably a good idea anyway to filter out special chars when determining the file name (so in this case it could be `username_password.go`). I wouldn't let that be the reason to change it.

Reply:
I decided that "User Credentials" may actually be a better choice than "Username and Password".

Although this is quite nitpicky, using "Username and Password" would result in the Snowflake credentials being imported as "Snowflake Username and Password" which isn't completely accurate and didn't look great to me.

<img width="474" alt="Screenshot 2023-02-09 at 10 29 02 AM" src="https://user-images.githubusercontent.com/53918631/217905403-847f2531-ca8e-4510-adb1-dab296f75340.png">

"User Credentials" seems general enough to encompass credentials that require just a username and password, as well as credentials that require other fields to authenticate as well, like `account` in the case of this SnowSQL CLI. 

<img width="474" alt="Screenshot 2023-02-09 at 10 33 21 AM" src="https://user-images.githubusercontent.com/53918631/217905906-6f0dc5e8-cbe3-46a9-a1a9-79570d7720db.png">

Reply:
By the way I noticed that the credential types get published to the Go package here: https://pkg.go.dev/github.com/1Password/shell-plugins@v0.0.0-20230126100958-d4822d5ebb3d/sdk/schema/credname. If we make changes to the credential types, like I did in this PR, is there any additional action we need to take so that it gets reflected on the page I linked or would the changes get mirrored automatically once the PR is merged? cc: @SimonBarendse

Reply:
> is there any additional action we need to take so that it gets reflected on the page I linked or would the changes get mirrored automatically once the PR is merged?

I managed to reflect the [latest version of this repo on pkg.go.dev](https://pkg.go.dev/github.com/1Password/shell-plugins?tab=versions) by running:

` GOPROXY=proxy.golang.org go list -m github.com/1Password/shell-plugins@5c8ee89` where 5c8ee89 is the latest commit short hash, to obtain the canonical version: `github.com/1Password/shell-plugins v0.0.0-20230210102903-5c8ee89b020f`

Then I ran `curl https://sum.golang.org/lookup/github.com/1password/shell-plugins@v0.0.0-20230210102903-5c8ee89b020f` to force pkg.go.dev to obtain the hash for that particular version. If the version doesn't exist, it will fetch it - and then hash it.

Reply:
> `User Credentials` seems general enough to encompass credentials that require just a username and password, as well as credentials that require other fields to authenticate as well, like account in the case of this SnowSQL CLI. 

In this case, I would advise on `User Login Details` as it is more specific. It is a synonym to username +password but can also encompass additional credentials. I am agreeing with the fact that here we do not only deal with username and password but also the `account` field is required for auth

Reply:
@AndyTitu Changed to `Login Details` per your suggestion.

Reply:
This is good with me!